### PR TITLE
Citation boilerplate

### LINKED ISSUE #713: Add excerpts of how to cite fmriprep *AND* dependencies
Meanwhile we can't implement this in a dynamic fashion with duecredit, we need to credit the authors of the huge amounts of software that fmriprep uses.

Update the documentation with these literate descriptions of references.

### PR DESCRIPTION
**Pull request by chrisfilo**
Closes #713 and #600

### CODE REVIEW

**Comment by chrisfilo** [comment]
https://427-66973489-gh.circle-artifacts.com/0/home/ubuntu/docs/citing.html

File: docs/citing.rst [hunk 1]
```diff
@@ -54,61 +54,61 @@ we recommend to include in your paper.
...
   </p>
   <p>
-  11. 	Jenkinson M. Fast, automated, N-dimensional phase-unwrapping algorithm. Magn Reson Med [Internet]. 2003 Jan;49(1):193–7. Available from: http://dx.doi.org/10.1002/mrm.10354
+  11. 	Jenkinson M. Fast, automated, N-dimensional phase-unwrapping algorithm. Magn Reson Med [Internet]. 2003 Jan;49(1):193–7. Available from: <a href="http://dx.doi.org/10.1002/mrm.10354">http://dx.doi.org/10.1002/mrm.10354</a>
   </p>
   <p>
-  12. 	Huntenburg JM. Evaluating nonlinear coregistration of BOLD EPI and T1w images [Internet]. Freie Universität Berlin; 2014. Available from: http://pubman.mpdl.mpg.de/pubman/item/escidoc:2327525/component/escidoc:2327523/master_thesis_huntenburg_4686947.pdf
+  12. 	Huntenburg JM. Evaluating nonlinear coregistration of BOLD EPI and T1w images [Internet]. Freie Universität Berlin; 2014. Available from: <a href="http://pubman.mpdl.mpg.de/pubman/item/escidoc:2327525/component/escidoc:2327523/master_thesis_huntenburg_4686947.pdf">http://pubman.mpdl.mpg.de/pubman/item/escidoc:2327525/component/escidoc:2327523/master_thesis_huntenburg_4686947.pdf</a>
```
Comment (effigies) [inline docs/citing.rst]: https://hdl.handle.net/11858/00-001M-0000-002B-1CB5-A

File: docs/citing.rst [hunk 2]
```diff
@@ -54,61 +54,61 @@ we recommend to include in your paper.
...
   </p>
   <p>
-  13. 	Wang S, Peterson DJ, Gatenby JC, Li W, Grabowski TJ, Madhyastha TM. Evaluation of Field Map and Nonlinear Registration Methods for Correction of Susceptibility Artifacts in Diffusion MRI. Front Neuroinform [Internet]. 2017 [cited 2017 Feb 21];11. Available from: http://journal.frontiersin.org/article/10.3389/fninf.2017.00017/full
+  13. 	Wang S, Peterson DJ, Gatenby JC, Li W, Grabowski TJ, Madhyastha TM. Evaluation of Field Map and Nonlinear Registration Methods for Correction of Susceptibility Artifacts in Diffusion MRI. Front Neuroinform [Internet]. 2017 [cited 2017 Feb 21];11. Available from: <a href="http://journal.frontiersin.org/article/10.3389/fninf.2017.00017/full">http://journal.frontiersin.org/article/10.3389/fninf.2017.00017/full</a>
```
Comment (effigies) [inline docs/citing.rst]: DOI: https://doi.org/10.3389/fninf.2017.00017

File: docs/citing.rst [hunk 3]
```diff
@@ -6,35 +6,110 @@
...
+  <p style="font-style: italic;">
+  For more details of the pipeline see <a href="http://fmriprep.readthedocs.io/en/1.0.0-rc6/workflows.html">http://fmriprep.readthedocs.io/en/1.0.0-rc6/workflows.html</a>.
+  </p>
+  <p>
+  1. 	Esteban O, Blair R, Markiewicz CJ, Berleant SL, Moodie C, Ma F, Isik AI, Erramuzpe A, Kent JD, Goncalves M, Poldrack RA, Gorgolewski KJ. poldracklab/fmriprep: 1.0.0-rc6 [Internet]. 2017. Available from: <a href="https://zenodo.org/record/1009184">https://zenodo.org/record/1009184</a>
```
Comment (effigies) [inline docs/citing.rst]: I think most people will prefer to cite DOIs: https://doi.org/10.5281/zenodo.1009184

File: docs/citing.rst [hunk 4]
```diff
@@ -6,35 +6,110 @@
...
+  </p>
+  <p>
+  2. 	Gorgolewski K, Burns CD, Madison C, Clark D, Halchenko YO, Waskom ML, Ghosh SS. Nipype: a flexible, lightweight and extensible neuroimaging data processing framework in python. Front Neuroinform [Internet]. 2011 Aug 22;5(August):13. Available from: <a href="http://dx.doi.org/10.3389/fninf.2011.00013">http://dx.doi.org/10.3389/fninf.2011.00013</a>
```
Comment (effigies) [inline docs/citing.rst]: I would generally prefer `https://doi.org/` to `http://dx.doi.org`. (Applies to several entries.)

File: docs/citing.rst [hunk 5]
```diff
@@ -6,35 +6,110 @@
...
+  </p>
+  <p>
+  4. 	Tustison NJ, Avants BB, Cook PA, Zheng Y, Egan A, Yushkevich PA, Gee JC. N4ITK: improved N3 bias correction. IEEE Trans Med Imaging [Internet]. 2010 Jun;29(6):1310–20. Available from: <a href="http://dx.doi.org/10.1109/TMI.2010.2046908">http://dx.doi.org/10.1109/TMI.2010.2046908</a>
+  </p>
+  <p>
+  5. 	Dale A, Fischl B, Sereno MI. Cortical Surface-Based Analysis: I. Segmentation and Surface Reconstruction. Neuroimage. 1999;9(2):179–94.
+  </p>
+  <p>
+  6. 	Fonov VS, Evans AC, McKinstry RC, Almli CR, Collins DL. Unbiased nonlinear average age-appropriate brain templates from birth to adulthood. NeuroImage; Amsterdam [Internet]. 2009 Jul 1;47:S102. Available from: <a href="https://search.proquest.com/docview/1833173019">https://search.proquest.com/docview/1833173019</a>
```
Comment (effigies) [inline docs/citing.rst]: DOI: https://doi.org/10.1016/S1053-8119(09)70884-5

File: docs/citing.rst [hunk 6]
```diff
@@ -6,35 +6,110 @@
...
+  </p>
+  <p>
+  4. 	Tustison NJ, Avants BB, Cook PA, Zheng Y, Egan A, Yushkevich PA, Gee JC. N4ITK: improved N3 bias correction. IEEE Trans Med Imaging [Internet]. 2010 Jun;29(6):1310–20. Available from: <a href="http://dx.doi.org/10.1109/TMI.2010.2046908">http://dx.doi.org/10.1109/TMI.2010.2046908</a>
+  </p>
+  <p>
+  5. 	Dale A, Fischl B, Sereno MI. Cortical Surface-Based Analysis: I. Segmentation and Surface Reconstruction. Neuroimage. 1999;9(2):179–94.
```
Comment (effigies) [inline docs/citing.rst]: DOI: https://doi.org/10.1006/nimg.1998.0395

File: docs/citing.rst [hunk 7]
```diff
@@ -6,35 +6,110 @@
...
+  </p>
+  <p>
+  6. 	Fonov VS, Evans AC, McKinstry RC, Almli CR, Collins DL. Unbiased nonlinear average age-appropriate brain templates from birth to adulthood. NeuroImage; Amsterdam [Internet]. 2009 Jul 1;47:S102. Available from: <a href="https://search.proquest.com/docview/1833173019">https://search.proquest.com/docview/1833173019</a>
+  </p>
+  <p>
+  7. 	Avants BB, Epstein CL, Grossman M, Gee JC. Symmetric diffeomorphic image registration with cross-correlation: evaluating automated labeling of elderly and neurodegenerative brain. Med Image Anal [Internet]. 2008 Feb;12(1):26–41. Available from: <a href="http://dx.doi.org/10.1016/j.media.2007.06.004">http://dx.doi.org/10.1016/j.media.2007.06.004</a>
+  </p>
+  <p>
+  8. 	Jenkinson M, Bannister P, Brady M, Smith S. Improved optimization for the robust and accurate linear registration and motion correction of brain images. Neuroimage [Internet]. 2002 Oct;17(2):825–41. Available from: <a href="https://www.ncbi.nlm.nih.gov/pubmed/12377157">https://www.ncbi.nlm.nih.gov/pubmed/12377157</a>
```
Comment (effigies) [inline docs/citing.rst]: DOI: https://doi.org/10.1006/nimg.2002.1132

File: docs/citing.rst [hunk 8]
```diff
@@ -6,35 +6,110 @@
...
+  </p>
+  <p>
+  9. 	Andersson JLR, Skare S, Ashburner J. How to correct susceptibility distortions in spin-echo echo-planar images: application to diffusion tensor imaging. Neuroimage [Internet]. 2003 Oct;20(2):870–88. Available from: <a href="http://dx.doi.org/10.1016/S1053-8119(03)00336-7">http://dx.doi.org/10.1016/S1053-8119(03)00336-7</a>
+  </p>
+  <p>
+  10. 	Cox RW. AFNI: software for analysis and visualization of functional magnetic resonance neuroimages. Comput Biomed Res [Internet]. 1996 Jun;29(3):162–73. Available from: <a href="http://www.ncbi.nlm.nih.gov/pubmed/8812068">http://www.ncbi.nlm.nih.gov/pubmed/8812068</a>
```
Comment (effigies) [inline docs/citing.rst]: DOI: https://doi.org/10.1006/cbmr.1996.0014

File: docs/citing.rst [hunk 9]
```diff
@@ -6,35 +6,110 @@
...
+  </p>
+  <p>
+  11. 	Jenkinson M. Fast, automated, N-dimensional phase-unwrapping algorithm. Magn Reson Med [Internet]. 2003 Jan;49(1):193–7. Available from: <a href="http://dx.doi.org/10.1002/mrm.10354">http://dx.doi.org/10.1002/mrm.10354</a>
+  </p>
+  <p>
+  12. 	Huntenburg JM. Evaluating nonlinear coregistration of BOLD EPI and T1w images [Internet]. Freie Universität Berlin; 2014. Available from: <a href="http://pubman.mpdl.mpg.de/pubman/item/escidoc:2327525/component/escidoc:2327523/master_thesis_huntenburg_4686947.pdf">http://pubman.mpdl.mpg.de/pubman/item/escidoc:2327525/component/escidoc:2327523/master_thesis_huntenburg_4686947.pdf</a>
```
Comment (effigies) [inline docs/citing.rst]: This appears to be the preferred link (one click away from the actual PDF): http://hdl.handle.net/11858/00-001M-0000-002B-1CB5-A

File: docs/citing.rst [hunk 10]
```diff
@@ -6,35 +6,110 @@
...
+  </p>
+  <p>
+  13. 	Wang S, Peterson DJ, Gatenby JC, Li W, Grabowski TJ, Madhyastha TM. Evaluation of Field Map and Nonlinear Registration Methods for Correction of Susceptibility Artifacts in Diffusion MRI. Front Neuroinform [Internet]. 2017 [cited 2017 Feb 21];11. Available from: <a href="http://journal.frontiersin.org/article/10.3389/fninf.2017.00017/full">http://journal.frontiersin.org/article/10.3389/fninf.2017.00017/full</a>
```
Comment (effigies) [inline docs/citing.rst]: DOI: https://doi.org/10.3389/fninf.2017.00017

**Comment by chrisfilo** [comment]
Ok - this is ready for review. I added dynamic fetching of the Zenodo reference (made a small library for it - planning to reuse it in other projects).

**Comment by chrisfilo** [comment]
Library: https://github.com/chrisfilo/zenodo.js

File: docs/citing.rst [hunk 11]
```diff
@@ -6,35 +6,159 @@
...
+
+
+.. raw:: html
+
+   <script language="javascript">
+   var version = '1.0.0-rc6';
+   function fillCitation(){
+      $('#fmriprep_version').text(version);
+      $('#workflow_url').text('http://fmriprep.readthedocs.io/en/' + version + '/workflows.html');
+      $('#workflow_url').attr('href', 'http://fmriprep.readthedocs.io/en/' + version + '/workflows.html');
```
Comment (effigies) [inline docs/citing.rst]: https

File: docs/citing.rst [hunk 12]
```diff
@@ -6,35 +6,159 @@
...
+
+      function cb(err, zenodoID) {
+         getCitation(zenodoID, 'vancouver-brackets-no-et-al', function(err, citation) {
+            $('#fmriprep_citation').text(citation);
+         });
+         getDOI(zenodoID, function(err, DOI) {
+            $('#fmriprep_doi_url').text('http://doi.org/' + DOI);
+            $('#fmriprep_doi_url').attr('href', 'http://doi.org/' + DOI);
```
Comment (effigies) [inline docs/citing.rst]: https

File: docs/citing.rst [hunk 13]
```diff
@@ -6,35 +6,159 @@
...
+   }
+
+   function toggle() {
+        var controlElementsIds = ["freesurfer","slicetime", "AROMA", "ss_template", "SDC"];
+        var controlElementsIdsLength = controlElementsIds.length;
+
+        for (var i = 0; i < controlElementsIdsLength; i++) {
+        	controlElement = document.getElementById(controlElementsIds[i])
+
+        	if (controlElement.checked == null){
```
Comment (effigies) [inline docs/citing.rst]: Mixed tabs and spaces in this line and the one above it.

File: docs/citing.rst [hunk 14]
```diff
@@ -6,35 +6,159 @@
...
-      url          = {https://doi.org/10.5281/zenodo.996169}
-    }
+Select which options you have run FMRIPREP with to generate custom language
+we recommend to include in your paper.
+
+
+.. raw:: html
+
+   <script language="javascript">
+   var version = '1.0.0-rc6';
```
Comment (effigies) [inline docs/citing.rst]: I'd be inclined to leave this as `'latest'`. Anybody who's building docs manually is likely to be near HEAD and putting in a specific version makes this a place that needs updating.

File: docs/citing.rst [hunk 15]
```diff
@@ -6,35 +6,159 @@
...
+  <p style="font-style: italic;">
+  For more details of the pipeline see <a id="workflow_url" href="http://fmriprep.readthedocs.io/en/latest/workflows.html">http://fmriprep.readthedocs.io/en/latest/workflows.html</a>.
+  </p>
+  <p>
+  <span id="fmriprep_citation">FMRIPREP</span> Available from: <a id="fmriprep_doi_url" href="https://doi.org/10.5281/zenodo.852659">https://doi.org/10.5281/zenodo.852659</a>
+  <img src onerror='fillCitation()'>
+  </p>
+  <p>
+  2. 	Gorgolewski K, Burns CD, Madison C, Clark D, Halchenko YO, Waskom ML, Ghosh SS. Nipype: a flexible, lightweight and extensible neuroimaging data processing framework in python. Front Neuroinform [Internet]. 2011 Aug 22;5(August):13. Available from: <a href="http://doi.org/10.3389/fninf.2011.00013">http://doi.org/10.3389/fninf.2011.00013</a>
```
Comment (effigies) [inline docs/citing.rst]: Sorry, wasn't clear. The main thing I was suggesting with the DOI URLs was the **s** in `https`, more than just dropping the `dx`. (I suspect https://dx.doi.org works, but figure shorter URLs are better.)

File: docs/conf.py [hunk 16]
```diff
@@ -308,3 +308,8 @@
 intersphinx_mapping = {'https://docs.python.org/': None}
 
 suppress_warnings = ["image.nonlocal_uri"]
+
+
+def setup(app):
+    # We need this for the boilerplate script
+    app.add_javascript("https://cdn.rawgit.com/chrisfilo/zenodo.js/v0.1/zenodo.js")
```
Comment (effigies) [inline docs/conf.py]: Newline at end of file.

**Comment by chrisfilo** [comment]
For some reason GitHub does not display the new line, but this should be done now.

File: circle.yml [hunk 17]
```diff
@@ -35,6 +35,7 @@ dependencies:
   override:
     - sed -i -E "s/(__version__ = )'[A-Za-z0-9.-]+'/\1'$CIRCLE_TAG'/" fmriprep/info.py
```
Comment (oesteban) [inline circle.yml]: Can we join together these 3 replacements in only one?:

```Bash
      - |
        if [[ "$CIRCLE_TAG" != "" ]]; then
            sed -i -E "s/(__version__ = )'[A-Za-z0-9.-]+'/\1'$CIRCLE_TAG'/" fmriprep/info.py
            sed -i -E "s/(__version__ = )'[A-Za-z0-9.-]+'/\1'$CIRCLE_TAG'/" wrapper/fmriprep_docker.py
            sed -i -E "s/(var version = )'[A-Za-z0-9.-]+'/\1'$CIRCLE_TAG'/" docs/citing.rst
        fi
```

File: docs/citing.rst [hunk 18]
```diff
@@ -6,35 +6,159 @@
...
+      $('#fmriprep_version').text(version);
+      $('#workflow_url').text('https://fmriprep.readthedocs.io/en/' + version + '/workflows.html');
+      $('#workflow_url').attr('href', 'https://fmriprep.readthedocs.io/en/' + version + '/workflows.html');
+
+      function cb(err, zenodoID) {
+         getCitation(zenodoID, 'vancouver-brackets-no-et-al', function(err, citation) {
+            $('#fmriprep_citation').text(citation);
+         });
+         getDOI(zenodoID, function(err, DOI) {
+            $('#fmriprep_doi_url').text('https://doi.org/' + DOI);
```
Comment (oesteban) [inline docs/citing.rst]: I usually prefer to hide the https://doi.org/ on DOIs and just show doi:[some-doi-without-http-doi-org](https://doi.org). But it is only a preference.

File: docs/citing.rst [hunk 19]
```diff
@@ -6,35 +6,159 @@
...
+  <p style="font-style: italic;">
+  For more details of the pipeline see <a id="workflow_url" href="http://fmriprep.readthedocs.io/en/latest/workflows.html">http://fmriprep.readthedocs.io/en/latest/workflows.html</a>.
+  </p>
+  <p>
+  <span id="fmriprep_citation">FMRIPREP</span> Available from: <a id="fmriprep_doi_url" href="https://doi.org/10.5281/zenodo.852659">https://doi.org/10.5281/zenodo.852659</a>
+  <img src onerror='fillCitation()'>
+  </p>
+  <p>
+  2. 	Gorgolewski K, Burns CD, Madison C, Clark D, Halchenko YO, Waskom ML, Ghosh SS. Nipype: a flexible, lightweight and extensible neuroimaging data processing framework in python. Front Neuroinform [Internet]. 2011 Aug 22;5(August):13. Available from: <a href="https://doi.org/10.3389/fninf.2011.00013">https://doi.org/10.3389/fninf.2011.00013</a>
```
Comment (oesteban) [inline docs/citing.rst]: As before, I usually prefer:

```
doi:<a href="https://doi.org/10.3389/fninf.2011.00013">10.3389/fninf.2011.00013</a>
```
over:
```
 Available from: <a href="https://doi.org/10.3389/fninf.2011.00013">https://doi.org/10.3389/fninf.2011.00013</a>
```

File: docs/citing.rst [hunk 20]
```diff
@@ -6,35 +6,159 @@
...
+   </select><br />
+   With AROMA: <input id="AROMA" type="checkbox" onclick="toggle();"/><br />
+   Skullstrip template: <select id="ss_template" onclick="toggle();">
+     <option value="OASIS" selected="true">OASIS</option>
+     <option value="NKI">NKI</option>
+   </select><br />
+   With slicetime correction: <input id="slicetime" type="checkbox" checked="true" onclick="toggle();"/><br />
+   </p>
+
+   <p style="font-style: italic;">Results included in this manuscript come from preprocessing performed using FMRIPREP version <span id="fmriprep_version">latest</span> [1] a Nipype [2,3] based tool. Each T1 weighted volume was corrected for bias field using N4BiasFieldCorrection v2.1.0 [4] and skullstripped using antsBrainExtraction.sh v2.1.0 (using
```
Comment (oesteban) [inline docs/citing.rst]: Can we add links to the references? ([1, 2, etc])

**Comment by effigies** [comment]
Ah, sorry. I'm okay with all of those changes, if you want to make a new PR.

**Comment by effigies** [comment]
If you do make a new one, we may want to add DOI:[10.1016/j.neuroimage.2007.12.025](https://dx.doi.org/10.1016/j.neuroimage.2007.12.025) for T2-based pial refinement (possibly another checkbox next to FreeSurfer?).

[This](https://www.mail-archive.com/freesurfer@nmr.mgh.harvard.edu/msg55536.html) was just posted on the mailing list this morning. I wasn't aware there was any citation for T2 refinement before.

**Comment by chrisfilo** [comment]
Please open issues for all the fixes/enhancements you mentioned.

**Pull request merged**